### PR TITLE
Finish PostgreSQLErrorResponsePacket

### PR DESCRIPTION
**Pull request by avalon566**

### CODE REVIEW

File: sharding-proxy/sharding-proxy-transport/sharding-proxy-transport-postgresql/src/main/java/org/apache/shardingsphere/shardingproxy/transport/postgresql/packet/generic/PostgreSQLErrorResponsePacket.java [hunk 1]
```diff
@@ -22,20 +22,78 @@
 import org.apache.shardingsphere.shardingproxy.transport.postgresql.packet.command.PostgreSQLCommandPacketType;
 import org.apache.shardingsphere.shardingproxy.transport.postgresql.payload.PostgreSQLPacketPayload;
 
+import java.util.HashMap;
+import java.util.Map;
+
 /**
  * Error response packet for PostgreSQL.
+ * https://www.postgresql.org/docs/12/protocol-message-formats.html
```
[the remark this answers is not in the document]
Comment (tuohai666) [reply sharding-proxy/sharding-proxy-transport/sharding-proxy-transport-postgresql/src/main/java/org/apache/shardingsphere/shardingproxy/transport/postgresql/packet/generic/PostgreSQLErrorResponsePacket.java]: Chage to: 

image

File: sharding-proxy/sharding-proxy-transport/sharding-proxy-transport-postgresql/src/main/java/org/apache/shardingsphere/shardingproxy/transport/postgresql/packet/generic/PostgreSQLErrorResponsePacket.java [hunk 2]
```diff
@@ -22,20 +22,78 @@
...
  *
  * @author zhangyonglun
+ * @author avalon566
  */
 public final class PostgreSQLErrorResponsePacket implements PostgreSQLPacket {
-    
+
+    public static final char FIELD_TYPE_SEVERITY = 'S';
+
+    /**
```
Comment (tuohai666) [inline sharding-proxy/sharding-proxy-transport/sharding-proxy-transport-postgresql/src/main/java/org/apache/shardingsphere/shardingproxy/transport/postgresql/packet/generic/PostgreSQLErrorResponsePacket.java]: remove the following 3 lines

File: sharding-proxy/sharding-proxy-transport/sharding-proxy-transport-postgresql/src/main/java/org/apache/shardingsphere/shardingproxy/transport/postgresql/packet/generic/PostgreSQLErrorResponsePacket.java [hunk 3]
```diff
@@ -22,20 +22,78 @@
...
-        payload.writeInt1(errorCode);
+        for (Map.Entry<Character, String> each : fields.entrySet()) {
+            payload.writeInt1(each.getKey());
+            payload.writeStringNul(each.getValue());
+        }
+        payload.writeInt1(0);
+    }
+
+    /**
+     * add field.
```
[the remark this answers is not in the document]
Comment (avalon566) [reply sharding-proxy/sharding-proxy-transport/sharding-proxy-transport-postgresql/src/main/java/org/apache/shardingsphere/shardingproxy/transport/postgresql/packet/generic/PostgreSQLErrorResponsePacket.java]: I had run checkstyle, no problem detected. [checkstyle file](https://github.com/apache/incubator-shardingsphere/blob/dev/src/resources/checkstyle.xml)

File: sharding-proxy/sharding-proxy-transport/sharding-proxy-transport-postgresql/src/main/java/org/apache/shardingsphere/shardingproxy/transport/postgresql/packet/generic/PostgreSQLErrorResponsePacket.java (hunk 2 above)
Comment (avalon566) [reply sharding-proxy/sharding-proxy-transport/sharding-proxy-transport-postgresql/src/main/java/org/apache/shardingsphere/shardingproxy/transport/postgresql/packet/generic/PostgreSQLErrorResponsePacket.java]: I don't quite understand?

File: sharding-proxy/sharding-proxy-transport/sharding-proxy-transport-postgresql/src/main/java/org/apache/shardingsphere/shardingproxy/transport/postgresql/packet/generic/PostgreSQLErrorResponsePacket.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (tuohai666) [reply sharding-proxy/sharding-proxy-transport/sharding-proxy-transport-postgresql/src/main/java/org/apache/shardingsphere/shardingproxy/transport/postgresql/packet/generic/PostgreSQLErrorResponsePacket.java]: But the first letter should be capital: add -> Add.

File: sharding-proxy/sharding-proxy-transport/sharding-proxy-transport-postgresql/src/main/java/org/apache/shardingsphere/shardingproxy/transport/postgresql/packet/generic/PostgreSQLErrorResponsePacket.java (hunk 2 above)
Comment (tuohai666) [reply sharding-proxy/sharding-proxy-transport/sharding-proxy-transport-postgresql/src/main/java/org/apache/shardingsphere/shardingproxy/transport/postgresql/packet/generic/PostgreSQLErrorResponsePacket.java]: I comment on line 39, I mean remove 3 lines from 39, i.e. remove line 39, 40, 41.

Comment (avalon566) [reply sharding-proxy/sharding-proxy-transport/sharding-proxy-transport-postgresql/src/main/java/org/apache/shardingsphere/shardingproxy/transport/postgresql/packet/generic/PostgreSQLErrorResponsePacket.java]: No comments required?

File: sharding-proxy/sharding-proxy-transport/sharding-proxy-transport-postgresql/src/main/java/org/apache/shardingsphere/shardingproxy/transport/postgresql/packet/generic/PostgreSQLErrorResponsePacket.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (avalon566) [reply sharding-proxy/sharding-proxy-transport/sharding-proxy-transport-postgresql/src/main/java/org/apache/shardingsphere/shardingproxy/transport/postgresql/packet/generic/PostgreSQLErrorResponsePacket.java]: I'll pay attention next time.

**Comment by avalon566** [comment]
Sorry, I've fixed it.

**Comment by tuohai666** [comment]
The CI failed. Please double check the codes before push, thanks.

[ERROR] Failed to execute goal org.apache.rat:apache-rat-plugin:0.12:check (default) on project sharding-proxy-transport-postgresql: Too many files with unapproved license: 1 See RAT report in: /home/jenkins/jenkins-slave/workspace/shardingsphere-ci/sharding-proxy/sharding-proxy-transport/sharding-proxy-transport-postgresql/target/rat.txt -> [Help 1]

**Comment by avalon566** [comment]
Sorry, I will compile locally first next time.

File: sharding-proxy/sharding-proxy-transport/sharding-proxy-transport-postgresql/src/main/java/org/apache/shardingsphere/shardingproxy/transport/postgresql/packet/generic/PostgreSQLErrorResponsePacket.java [hunk 4]
```diff
@@ -22,20 +22,75 @@
...
-    
-    private final char errorCode = '\0';
-    
+
+    private final Map<Character, String> fields = new HashMap<>();
+
     @Override
     public void write(final PostgreSQLPacketPayload payload) {
-        payload.writeInt1(errorCode);
+        for (Map.Entry<Character, String> each : fields.entrySet()) {
```
Comment (tuohai666) [inline sharding-proxy/sharding-proxy-transport/sharding-proxy-transport-postgresql/src/main/java/org/apache/shardingsphere/shardingproxy/transport/postgresql/packet/generic/PostgreSQLErrorResponsePacket.java]: Sorry, I didn't notice this before. Please:
import java.util.Map.Entry;
and
Map.Entry -> Entry

**Comment by avalon566** [comment]
Thanks for review. @tuohai666

**Comment by tuohai666** [comment]
My pleasure.

**Pull request merged**